wallet/src/program_facades/amm.rs

```diff
@@ -0,0 +1,451 @@
+        };
+
+        let definition_token_a_id = TokenHolding::parse(&user_a_acc.data)
+            .ok_or(ExecutionFailureKind::AccountDataError(user_holding_a))?
+            .definition_id;
+        let definition_token_b_id = TokenHolding::parse(&user_b_acc.data)
+            .ok_or(ExecutionFailureKind::AccountDataError(user_holding_b))?
+            .definition_id;
+
+        let amm_pool =
+            compute_pool_pda(amm_program_id, definition_token_a_id, definition_token_b_id);
+        let vault_holding_a = compute_vault_pda(amm_program_id, amm_pool, definition_token_a_id);
+        let vault_holding_b = compute_vault_pda(amm_program_id, amm_pool, definition_token_b_id);
+
+        let account_ids = vec![
+            amm_pool,
+            vault_holding_a,
+            vault_holding_b,
+            user_holding_a,
+            user_holding_b,
+        ];
+
+        let account_id_auth;
+
+        // Checking, which account are associated with TokenDefinition
+        let token_holder_acc_a = self
+            .0
+            .get_account_public(user_holding_a)
+            .await
+            .map_err(|_| ExecutionFailureKind::SequencerError)?;
+        let token_holder_acc_b = self
+            .0
+            .get_account_public(user_holding_b)
+            .await
+            .map_err(|_| ExecutionFailureKind::SequencerError)?;
+
+        let token_holder_a = TokenHolding::parse(&token_holder_acc_a.data)
+            .ok_or(ExecutionFailureKind::AccountDataError(user_holding_a))?;
+        let token_holder_b = TokenHolding::parse(&token_holder_acc_b.data)
+            .ok_or(ExecutionFailureKind::AccountDataError(user_holding_b))?;
+
+        if token_holder_a.definition_id == token_definition_id {
+            account_id_auth = user_holding_a;
+        } else if token_holder_b.definition_id == token_definition_id {
+            account_id_auth = user_holding_b;
+        } else {
+            return Err(ExecutionFailureKind::AccountDataError(token_definition_id));
+        }
+
+        let Ok(nonces) = self.0.get_accounts_nonces(vec![account_id_auth]).await else {
+            return Err(ExecutionFailureKind::SequencerError);
+        };
+
+        let Some(signing_key) = self
+            .0
+            .storage
+            .user_data
+            .get_pub_account_signing_key(&account_id_auth)
+        else {
+            return Err(ExecutionFailureKind::KeyNotFoundError);
+        };
+
+        let message = nssa::public_transaction::Message::try_new(
+            program.id(),
+            account_ids,
+            nonces,
+            instruction,
+        )
+        .unwrap();
+
+        let witness_set =
+            nssa::public_transaction::WitnessSet::for_message(&message, &[signing_key]);
+
+        let tx = nssa::PublicTransaction::new(message, witness_set);
+
+        Ok(self.0.sequencer_client.send_tx_public(tx).await?)
+    }
+
+    pub async fn send_add_liq(
```

Comment:
```suggestion
    pub async fn send_add_liquidity(
```

Let's stick to a full names especially for public api if possible